src/tools.cpp

```diff
@@ -106,9 +106,13 @@ std::tuple<char, std::string> zim::parseLongPath(const std::string& longPath)
 {
   /* Index of the namespace char; discard '/' from absolute paths */
   const unsigned int i = (longPath[0] == '/') ? 1 : 0;
-  if (i + 2 >= longPath.size() || longPath[i] == '/' || longPath[i+1] != '/')
+  if (i + 1 > longPath.size() || longPath[i] == '/' || (i + 2 <= longPath.size() && longPath[i+1] != '/'))
```

Comment:
The condition `(i + 2 <= longPath.size() && longPath[i+1] != '/')` will be a little easier to understand if rewritten as `(i + 1 < longPath.size() && longPath[i+1] != '/')` (since `i+1` is used as an index into `longPath` we check that it isn't out of bounds)